src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -487,7 +491,71 @@ private void updateBundleAndRecipientListWithResponses(
         bundle.put(question, responses);
         recipientList.put(question.getId(), recipients);
     }
-    
+
+    /**
+     * Removes instructors who are not displayed to students from the
+     * {@link FeedbackQuestionBundle}
+     * 
+     * @param bundle
+     *            {@link FeedbackQuestionsBundle} containing questions for the
+     *            session
+     */
+    private void filterOutHiddenInstructors(FeedbackSessionQuestionsBundle bundle) {
```

Comment:
The `updateBundleAndRecipientLIstWithResponses` (for student) only updates the `FeedbackSesssionQuestionsBundle` for one question. We need to perform database query many times if we get the list of hidden instructors in this method and there are many questions with instructor recipient type, otherwise the method has to take in an additional parameter (the list of hidden instructors).